plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointSchemaManager.java

```diff
@@ -114,10 +115,22 @@ public RowType getMetadataEntryType()
         return metadataEntryType;
     }
 
-    public RowType getAddEntryType(MetadataEntry metadataEntry, ProtocolEntry protocolEntry, boolean requireWriteStatsAsJson, boolean requireWriteStatsAsStruct, boolean usePartitionValuesParsed)
+    public RowType getAddEntryType(
+            MetadataEntry metadataEntry,
+            ProtocolEntry protocolEntry,
+            Optional<Function<String, Boolean>> addStatsColumnFilter,
```

Comment:
empty function is equivalent to an always-true function
we don't need Optional here